py/bitbox02/bitbox02/communication/bitbox_api_protocol.py

```diff
@@ -407,3 +441,27 @@ def get_info(self) -> Tuple[str, Platform, Union[BitBox02Edition, BitBoxBaseEdit
         unlocked_byte = response[0]
         unlocked = {0x00: False, 0x01: True}[unlocked_byte]
         return (version_str, platform, edition, unlocked)
+
+    def check_version(self) -> None:
+        """
+        Returns false if the device has an older firmware version than required
+        and the minimum required version. A check for the BitBoxBase is not implemented.
+        """
+        if self.edition == BitBox02Edition.MULTI:
+            if self.version < MIN_BITBOX02_MULTI_FIRMWARE_VERSION:
+                raise FirmwareVersionOutdatedException(
+                    self.version, MIN_BITBOX02_MULTI_FIRMWARE_VERSION
+                )
+            if self.version >= semver.VersionInfo(
+                MIN_BITBOX02_MULTI_FIRMWARE_VERSION.major + 1, 0, 0
+            ):
+                raise LibraryVersionOutdatedException(self.version)
+        elif self.edition == BitBox02Edition.BTCONLY:
```

Comment:
else raise exception (think of it like a switch which should always treat all possibilities)